### PR TITLE
Update links from Delegate Crash Course to Delegate Handbook

### PR DESCRIPTION
**Pull request by nsilvestri**
The Delegate Crash Course has been deprecated in favor of the Delegate Handbook. This PR updates existing references to the DCC to the Handbook where it is still relevant, or removes references to the DCC entirely where they are no longer needed.

### CODE REVIEW

**Comment by danieljames-dj** [comment]
Hi @nsilvestri, may I know by when are you planning to get this PR merged? Does WQAC has any deadline for the same?

The reason why I'm asking is, in /panel, Delegate Panel is going to be migrated to a separate react component. I've already raised the PR to migrate Delegate Panel: https://github.com/thewca/worldcubeassociation.org/pull/9332. Once that is merged, you should be able to update delegate-crash-course to delegate-handbook more easily probably.

**Comment by nsilvestri** [comment]
@danieljames-dj this PR will be pending the document being added to wca-documents, hopefully 2ish weeks? Don't let this PR be a blocker for making any changes to the panel, I can adapt to whatever changes happen in the meantime.

File: config/routes.rb
Comment (danieljames-dj) [inline config/routes.rb:412]: Only the RHS needs to be changed. The LHS is actually meant to be the link that existed earlier and doesn't exist now, but still supporting in case the users have them bookmarked.

File: app/views/results_submission/_results_submission_panel.html.erb
Comment (danieljames-dj) [inline app/views/results_submission/_results_submission_panel.html.erb:25]: 1. I tried opening this link, but it's not scrolling to the section in the handbook. Is this something that is supported in at least some browsers?
2. Having the same URL hardcoded in two places looks bit odd, but I cannot think of a way to solve it especially because there is a hash parameter. Just mentioning here so that you or somebody else will be able to find a solution.

**Review by danieljames-dj — COMMENTED** [review]
The code looks good to me. I've raised a comment at the place where I have a slight concern but not sure if it's solvable. I would like @gregorbg to have a look to give the final approval.

Thanks @nsilvestri for working on this. 🎉

File: app/views/results_submission/_results_submission_panel.html.erb
Comment (nsilvestri) [reply app/views/results_submission/_results_submission_panel.html.erb:25]: 1. Yes; on Firefox the link takes me straight to the section. It should be supported in other browses as AFAIK it's the same as heading tag hashes in URLs, but if it doesn't work in some browser configuration that's nbd to me since otherwise the link works and the section is explicitly called out here.
2. I'm not sure how you'd prefer to handle a constant like this across Rails code and React code, open to suggestions if it's a priority fix. Changing this around would also be easy in a subsequent commit if required.

**Comment by danieljames-dj** [comment]
I'm approving this for now, if any solution for the URL comes up, we can take it up later.

**Review by danieljames-dj — APPROVED** [review]

**Pull request merged**
